src/ape/api/projects.py

```diff
@@ -115,6 +115,40 @@ def _create_manifest(
 
         manifest.sources = cls._create_source_dict(source_paths, contracts_path)
         manifest.contract_types = contract_types
+
+        manifest.compilers = []
+        # [x for x in contract_types]
+        i = 0
+        for contract_type in contract_types:
+            suffix = source_paths[i].suffix
+            if cls.compiler_manager.registered_compilers[suffix].name:
+                if suffix == ".sol":
+                    manifest.compilers.append(
+                        {
+                            "name": cls.compiler_manager.registered_compilers[suffix].name,
+                            "version": [
+                                x
+                                for x in cls.compiler_manager.registered_compilers[
+                                    suffix
+                                ].get_version_map(source_paths)
+                            ][i],
+                            "settings": None,
+                            "contractTypes": contract_types[contract_type],
+                        }
+                    )
+                else:
+                    manifest.compilers.append(
+                        {
+                            "name": cls.compiler_manager.registered_compilers[suffix].name,
+                            "version": cls.compiler_manager.registered_compilers[
+                                suffix
+                            ].get_versions(source_paths),
+                            "settings": None,
+                            "contractTypes": contract_types[contract_type],
+                        }
+                    )
+            i += 1
```

Comment:
I don't think this will work here because it's a classmethod, I would try to move it to where it actually compiles the files

Reply:
it seems to mostly be working, I think I have to fix the iterator. I'll look to see how I can move it to where compilation is occurring if that seems like it would work better.

Reply:
If you use one of the managers in a classmethod I would suspect it may run into weird race conditions down the line

Reply:
I think it's fine to be here, goes along with the source dict logic, haven't seen any race conditions with it. Here's maybe a place to start exploring.

```python
    @classmethod
    def _create_manifest(
        cls,
        source_paths: List[Path],
        contracts_path: Path,
        contract_types: Dict[str, ContractType],
        name: Optional[str] = None,
        version: Optional[str] = None,
        initial_manifest: Optional[PackageManifest] = None,
    ) -> PackageManifest:
        manifest = initial_manifest or PackageManifest()

        if name:
            manifest.name = PackageName(name.lower())

        if version:
            manifest.version = version

        manifest.compilers = cls._create_compilers_list(source_paths, contract_types)
        manifest.sources = cls._create_source_dict(source_paths, contracts_path)
        manifest.contract_types = contract_types

        return manifest

    @classmethod
    def _create_compilers_list(
        cls,
        source_paths: List[Path],
        contract_types: Dict[str, ContractType],
    ):
        compiler_list: List[Compiler] = []

        for key, compiler in cls.compiler_manager.registered_compilers.items():
            compiler_list.append(
                Compiler(
                    name=compiler.name,
                    # TODO: Figure out if this should be str
                    version=str(compiler.get_versions(source_paths)),
                    # TODO: Find compiler settings
                    settings={},
                    # TODO: Figure out what file type/compiler for each contract_typeT
                    # TODO: You'll need to filter these to match the compiler used
                    # NOTE: Consider adding a compiler member to ContractType class?
                    # NOTE: Check data type, ethpm-types says List[str] not sure what this data should be example passes name of contract
                    contractTypes=[str(x) for x in contract_types.keys()],
                )
            )

        return compiler_list
```

Reply:
Oh yeah, forgot that the ethpm will be in there